# Sesión 10 · Entiende lo que resumen los números

Elige medidas útiles y reconoce lo que una muestra puede y no puede decir.

Duración: 150 minutos de clase + 45–60 minutos de tarea. Ejecuta en orden. Los ejemplos son independientes salvo indicación.

## Objetivos
- Comparar media y mediana.
- Interpretar dispersión y cuantiles.
- Distinguir asociación de causalidad.

## Centro: media y mediana

La media reparte el total por igual entre observaciones. La mediana es el valor central después de ordenar. Responden a preguntas distintas y reaccionan de forma diferente a extremos.

En ingresos [100, 100, 100, 100, 1000], la media supera la mayoría de observaciones. Mostrar también mediana y distribución evita describir mal el caso típico.

In [1]:
import pandas as pd

ingresos = pd.Series([100, 100, 100, 100, 1000])
print("Media:", ingresos.mean())
print("Mediana:", ingresos.median())
print("Total:", ingresos.sum())

Media: 280.0
Mediana: 100.0
Total: 1400


**Qué observar:** Media 280.0; mediana 100.0; total 1400.

**Recuerda:** Una medida no es mejor en todos los casos. Decide según la pregunta y explica qué representa.

## Dispersión y cuantiles

Dos grupos pueden tener la misma media y variaciones muy distintas. El rango compara extremos; la desviación estándar resume distancias respecto a la media en las mismas unidades.

Los cuantiles describen posiciones en la distribución. El rango intercuartílico, Q3 − Q1, resume la amplitud del 50 % central. Aquí usamos ddof=0 para describir el conjunto completo del ejemplo.

In [2]:
import pandas as pd

a = pd.Series([10, 10, 10])
b = pd.Series([0, 10, 20])
print(a.mean(), b.mean())
print(a.std(ddof=0), round(b.std(ddof=0), 2))
print(b.quantile([0.25, 0.5, 0.75]))

10.0 10.0
0.0 8.16
0.25     5.0
0.50    10.0
0.75    15.0
dtype: float64


**Qué observar:** Ambas medias son 10.0; desviaciones 0.0 y 8.16. Cuantiles: 5, 10 y 15.

**Recuerda:** Pandas usa ddof=1 por defecto para std; especifica la convención cuando sea importante para interpretar o comparar resultados.

## Asociación, muestras y límites

La correlación de Pearson describe asociación lineal entre variables numéricas. No prueba que una cause a la otra y puede cambiar por extremos o grupos mezclados.

Un conjunto de datos representa las observaciones que se recogieron. Si solo analizas clientes de una sucursal, no puedes asegurar que las conclusiones describan a todo el país.

In [3]:
import pandas as pd

datos = pd.DataFrame({"Publicidad": [1, 2, 3, 4], "Ventas": [10, 20, 30, 40]})
print(datos.corr())

            Publicidad  Ventas
Publicidad         1.0     1.0
Ventas             1.0     1.0


**Qué observar:** La correlación entre ambas columnas es 1.0 en este ejemplo construido.

**Recuerda:** La asociación perfecta de cuatro datos artificiales no demuestra un efecto causal de publicidad sobre ventas.

## Práctica
Intenta resolver antes de consultar las soluciones del aula web.

### ¿Qué es típico?

Calcula media y mediana de [10, 12, 13, 15, 100]. Explica la diferencia.

**Criterio de logro:** Media 30.0; mediana 13.0.

In [ ]:
# Escribe tu solución aquí.

### Misma media, distinta variación

Compara [5, 5, 5] y [1, 5, 9] con media y rango.

**Criterio de logro:** Medias 5; rangos 0 y 8.

In [ ]:
# Escribe tu solución aquí.

## Tarea · Un resumen que no engañe

Usa ingresos [1200, 1500, 1600, 1700, 1800, 9000].

1. Calcula tamaño, media, mediana, mínimo y máximo.
2. Calcula Q1, Q3 y rango intercuartílico.
3. Compara media y mediana con una explicación contextual.
4. Escribe una limitación de concluir a partir de solo seis observaciones.

Evaluación: corrección 50 %, explicación 30 %, orden y reproducibilidad 20 %.

In [ ]:
# Desarrolla tu tarea en esta celda y agrega las que necesites.

## Mi conclusión
Escribe tu interpretación y una limitación.